superset/db_engine_specs/presto.py

```diff
@@ -947,13 +948,20 @@ def latest_sub_partition(cls, table_name, schema, database, **kwargs):
         return df.to_dict()[field_to_return][0]
 
     @classmethod
-    @cache.memoize()
-    def get_function_names(cls, database: "Database") -> List[str]:
+    def get_function_names(
+        cls, database: "Database", schema: Optional[str]
+    ) -> List[str]:
         """
         Get a list of function names that are able to be called on the database.
         Used for SQL Lab autocomplete.
 
         :param database: The database to get functions for
+        :param schema: The schema to get functions for (N/A for Presto)
```

Comment:
Won't a schema still be passed for Presto even if it isn't being used? That would mean we'd still cache at the db + schema level for Presto (and hive) which isn't necessary.

Reply:
Memoization is only done for the nested function (see `_get_function_names` below), i.e. will not be cached per schema.